benchmark_runner/benchmark_operator/benchmark_operator_workloads.py

```diff
@@ -29,103 +29,27 @@ def stressng_pod(self, name: str = ''):
-            if self._run_type == 'test_ci':
-                self.__es_index = 'stressng-test-ci-results'
-            else:
-                self.__es_index = 'stressng-results'
-            self.__workload = self.stressng_vm.__name__.replace('_', '-')
-            environment_variables.environment_variables_dict['kind'] = 'vm'
-            self._oc.create_vm_sync(yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.stressng_vm.__name__}.yaml'), vm_name=f'{self.__workload}-workload')
-            self._oc.wait_for_initialized(label='app=stressng_workload', workload=self.__workload)
-            self._oc.wait_for_ready(label='app=stressng_workload', workload=self.__workload)
-            # Create vm log should be direct after vm is ready
-            vm_name = self._create_vm_log(labels=[self.__workload])
-            self.__status = self._oc.wait_for_vm_completed(workload=self.__workload, vm_name=vm_name)
-            self.__status = 'complete' if self.__status else 'failed'
-            # system metrics
-            if environment_variables.environment_variables_dict['system_metrics'] == 'True':
-                self.system_metrics_collector(workload=self.__workload)
-            # save run artifacts logs of benchmark-controller-manager and system-metrics
-            run_artifacts_url = self._create_run_artifacts(workload=self.__workload, pod=False)
-            # verify that data upload to elastic search
-            if self._es_host:
-                ids = self._verify_elasticsearch_data_uploaded(index=self.__es_index, uuid=self._oc.get_long_uuid(workload=self.__workload))
-                # update metadata
-                for id in ids:
-                    self._update_elasticsearch_index(index=self.__es_index, id=id, kind=self._environment_variables_dict.get('kind', ''), status=self.__status, run_artifacts_url=run_artifacts_url)
-            self._oc.delete_vm_sync(
-                yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.stressng_vm.__name__}.yaml'),
-                vm_name=f'{self.__workload}-workload')
-        except ElasticSearchDataNotUploaded as err:
-            self.tear_down_vm_after_error(yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.stressng_vm.__name__}.yaml'),
-                                          vm_name=f'{self.__workload}-workload')
-            raise err
-        except Exception as err:
-            # save run artifacts logs
-            run_artifacts_url = self._create_run_artifacts(workload=self.__workload, pod=False)
-            self._upload_to_elasticsearch(index=self.__es_index, kind=self._environment_variables_dict.get('kind', ''), status='failed', run_artifacts_url=run_artifacts_url)
-            self.tear_down_vm_after_error(yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.stressng_vm.__name__}.yaml'),
-                                          vm_name=f'{self.__workload}-workload')
-            raise err
+        run = StressngVM()
+        run.stressng_vm()
```

Comment:
This file should not contain workload-specific code.

Reply:
What it means ?

Reply:
@RobertKrawitz thx for this great feedback , remove all workloads from this module and create a dynamic calls.